service/debugger/debugger.go

```diff
@@ -0,0 +1,453 @@
+package debugger
+
+import (
+	"fmt"
+	"regexp"
+	"runtime"
+
+	"github.com/golang/glog"
+	sys "golang.org/x/sys/unix"
+
+	"github.com/derekparker/delve/proctl"
+	"github.com/derekparker/delve/service/api"
+)
+
+// Debugger provides a thread-safe DebuggedProcess service. Instances of
+// Debugger can be exposed by other services.
+type Debugger struct {
+	config       *Config
+	processFuncs chan func(*proctl.DebuggedProcess) error
```

Comment:
Remove the `error` return type here since error handling is done via the wrapper function pushed onto the channel in `withProcess`.